### PR TITLE
Added some common middleware ...

### PR DESCRIPTION
**Pull request by taphu**
... that is helpful to all of our projects.

### CODE REVIEW

File: src/Canteven/HTTP.hs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+{-# LANGUAGE LambdaCase #-}
```
Comment (potomak) [inline src/Canteven/HTTP.hs]: 😱

File: src/Canteven/HTTP.hs [hunk 2]
```diff
@@ -56,3 +78,99 @@ data DecodeResult e
...
+      try (lift (respond response)) >>= \case
+        Right ack -> return ack
+        Left err -> do
+          void $ logProblem err
+          throwM err
+
+    logProblem :: SomeException -> LoggingT IO UUID
+    logProblem err = do
+      uuid <- getUUID
+      logInfoN . pack
```
Comment (potomak) [inline src/Canteven/HTTP.hs]: Why not `logErrorN`?

Comment (taphu) [inline src/Canteven/HTTP.hs]: should be logError.

**Comment by potomak** [comment]
@taphu I think someone should add some info in the README about middlewares included in this library, maybe in a later PR.

**Comment by taphu** [comment]
> I think someone should add some info in the README about middlewares included in this library, maybe in a later PR.

Yes. absolutely.

**Pull request merged**